administrator/components/com_admin/script.php

```diff
@@ -282,6 +285,50 @@ protected function uninstallExtensions()
         }
     }
 
+    /**
+     * This method is for migration for old updatenotification system plugin migration to task.
+     *
+     * @param   \stdClass  $data  Object with `extension_id` and `params` of the extension
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    private function migrateUpdatenotificationPlugin($data)
+    {
+        if (!$data->enabled) {
+            return;
+        }
+
+        // Get the timeout for Joomla! updates, as configured in com_installer's component parameters
+        $component    = ComponentHelper::getComponent('com_installer');
+        $paramsc      = $component->getParams();
+        $cachetimeout = (int) $paramsc->get('cachetimeout', 6);
+        $params       = new Registry($data->params);
+        $lastrun      = (int) $pluginParams->get('lastrun', 0);
```

Comment:
![image](https://github.com/joomla/joomla-cms/assets/66922325/b93b1b64-1d85-45eb-aefb-ec06a118c814)

```suggestion
        $lastrun      = (int) $params->get('lastrun', 0);
```